pkg/bpf/bpf.go

```diff
@@ -303,8 +312,15 @@ func OpenOrCreateMap(path string, mapType int, keySize, valueSize, maxEntries ui
 
 		err = ObjPin(fd, path)
 		if err != nil {
+			// Need to close this fd since the next ObjGet
+			// returns a different fd.
+			ObjClose(fd)
 			return 0, isNewMap, err
 		}
+
+		// Need to close this fd since the next ObjGet
+		// returns a different fd.
+		ObjClose(fd)
```

Comment:
Instead of closing and calling `ObjGet` again, Daniel suggests to return the opened fd here.